### PR TITLE
http3: only use :protocol pseudo-header for Extended CONNECT

### PR DESCRIPTION
**Pull request by taoso**
The default value should be "HTTP/3.0".

### CODE REVIEW

**Comment by marten-seemann** [comment]
Can you explain the motivation behind this PR?

**Comment by taoso** [comment]
> Can you explain the motivation behind this PR?

hi @marten-seemann If the client fire a CONNECT request without :protocol, this req.Proto will be empty without this patch.

**Review by marten-seemann — COMMENTED** [review]
I think this PR is correct, but I'd to tighten the validation logic here a bit. The `:protocol` pseudo header is only defined for Extended Connect requests (RFC 9220). We should reject any non-Extended Connected request that sets this header field.

@taoso Do you want to update the PR? We'd also need a test case that tests that we correctly reject requests with `:protocol` header.

File: http3/headers.go [hunk 1]
```diff
@@ -124,11 +124,14 @@ func requestFromHeaders(headerFields []qpack.HeaderField) (*http.Request, error)
 		}
 	} else if len(hdr.Path) == 0 || len(hdr.Authority) == 0 || len(hdr.Method) == 0 {
 		return nil, errors.New(":path, :authority and :method must not be empty")
+	} else if len(hdr.Protocol) > 0 {
+		return nil, errors.New(":protocol must be empty")
```
Comment (marten-seemann) [inline http3/headers.go:128]: This branch needs to run `if !isExtendedConnected`. I think this is given by the current logic, but it's a bit subtle. I'd prefer to have this is in a separate if statement.

Comment (taoso) [reply http3/headers.go]: Fixed

**Review by marten-seemann — APPROVED** [review]
LGTM, thanks for pushing this through @taoso!

**Comment by taoso** [comment]
Hi @marten-seemann  Could this PR be merged?

**Pull request merged**